projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js

```diff
@@ -98,38 +100,81 @@ async function createOrUpdateComment( payload, octokit, issueReferences, issueCo
 	const { name: repo, owner } = repository;
 	const ownerLogin = owner.login;
 
-	const comment = `**Support References**
-	${ issueReferences
-		.map(
-			reference => `
-- [ ] ${ reference }`
-		)
-		.join( '' ) }
-
-	`;
-
-	const commentOpts = {
-		owner: ownerLogin,
-		repo,
-		body: comment,
-	};
-
 	const existingComment = await getListComment( issueComments );
 
 	// If there is a comment already, update it.
 	if ( existingComment !== 0 ) {
 		debug(
 			`gather-support-references: update comment ID ${ existingComment } with our new list of references.`
 		);
+
+		const {
+			data: { body: existingCommentBody },
+		} = await octokit.rest.issues.getComment( {
+			owner: ownerLogin,
+			repo,
+			comment_id: existingComment,
+		} );
+
+		// First, build a list of all references and whether they are checked or not.
+		const listWithStatusMatch = existingCommentBody.matchAll(
+			/(?:-\s\[(?<checked>\s|x)\]\s)(?<ticketId>[0-9]*-(?:hc|zen))/gm
+		);
+		const statusMatches = [ ...listWithStatusMatch ];
+
+		// Build a new array with only checked ticket references.
+		const checkedList = [];
+		for ( const referenceStatus of statusMatches ) {
+			const [ , checked, ticketId ] = referenceStatus;
+			if ( checked === 'x' ) {
+				checkedList.push( ticketId );
+			}
+		}
+
+		// Remove checked tickets from our existing list of issue references.
+		const uncheckedList = issueReferences.filter(
+			reference => ! checkedList.includes( reference )
+		);
+
+		// Build our comment body, with first the checked references, then the unchecked references.
+		const updatedComment = `**Support References**
+${ checkedList
+	.map(
+		checkedTicket => `
+- [x] ${ checkedTicket }`
+	)
+	.join( '' ) }${ uncheckedList
+			.map(
+				uncheckedTicket => `
+- [ ] ${ uncheckedTicket }`
+			)
+			.join( '' ) }
+
+`;
```

Comment:
I'd keep the one array of references, and only use `checkedList` to determine whether to `[ ]` or `[x]`. Something like this
```suggestion
		// Extract the checked ticket references.
		const checkedRefs = new Set();
		for ( const referenceStatus of statusMatches ) {
			const [ , checked, ticketId ] = referenceStatus;
			if ( checked === 'x' ) {
				checkedRefs.add( ticketId );
			}
		}

		// Build our comment body, with first the checked references, then the unchecked references.
		const updatedComment = `**Support References**
${ issueReferences
	.map(
		reference => `
- [${ checkedRefs.has( reference ) ? 'x' : ' ' }] ${ reference }`
	)
	.join( '' ) }

`;
```
That way is more likely to keep the references in a stable order, rather than reordering them randomly at some point after someone checks some off and then someone else adds a new comment.

Reply:
It does simplify things a bit, I like it. Done in f5ccd66e34bfe1bb601fd7566540449a3976555e